docs/blog/2025-01-20-linking-ts-in-monorepos.md

```diff
@@ -0,0 +1,666 @@
+
+With this setup, you can simplify the import in your application:
+
+```ts {% fileName="apps/myapp/src/index.ts" %}
+import { greet } from '@ts-monorepo-linking/lib-a';
+
+console.log(greet('World'));
+```
+
+### Observations
+
+**Dependency resolution:**  
+Path aliases eliminate the need for relative imports, resulting in a cleaner and more maintainable structure. If the underlying paths change, you only need to update the alias in `tsconfig.base.json`.
+
+> As a side note: While this article focuses on the build and type-checking phase, it's worth noting that running the transpiled TypeScript code directly wouldn't work out of the box. This is because TypeScript path aliases are purely a compile-time construct—they don't exist in the output JavaScript. To run the application, you'd need a runtime plugin or bundler (like Webpack, esbuild, or Vite) that can resolve these aliases to actual file paths.
+
+**Modularity:**  
+This approach doesn't change the modularity from the relative imports setup. TypeScript still treats the entire workspace as one large project, without enforcing strict boundaries between packages.
+
+**Performance :**  
+Path aliases don't improve performance compared to relative imports. The entire workspace is still treated as a single TypeScript project, so type checking and compilation remain unchanged. This approach focuses on maintainability and readability rather than optimizing performance.
+
+## Improving Performance with Project References
+
+Example: [Stackblitz](https://stackblitz.com/github/juristr/ts-monorepo-linking/tree/ts-proj-references-simple) - [Github](https://github.com/juristr/ts-monorepo-linking/tree/ts-proj-references-simple)
+
+TypeScript project references let you break a large TypeScript project into smaller, manageable units. This approach aligns with monorepo structures, allowing each package to act as its own TypeScript program while maintaining relationships between them.
+
+To use project references:
+
+- Add the `references` property in `tsconfig.json` files to point to dependent projects.
+- Enable `composite: true` in `compilerOptions` (this also enables `incremental` and `declaration` by default).
+- Use `tsc --build` (`tsc -b`) for compilation and type checking.
+
+> For a more deep-dive on TypeScript project references, make sure to check out our article << **TODO LINK HERE ZACK's POST**
+
+Our workspace structure still remains the same with the exception of adding another root-level `tsconfig.json`:
+
+```
+ts-monorepo-linking
+   ├─ apps
+   │  └─ myapp
+   │     ├─ src
+   │     │  └─ index.ts
+   │     └─ tsconfig.json
+   ├─ packages
+   │  └─ lib-a
+   │     ├─ src
+   │     │  └─ index.ts
+   │     └─ tsconfig.json
+   ├─ tsconfig.base.json
+   └─ tsconfig.json
+```
+
+This new `tsconfig.json` is the entry point for TypeScript project references, pointing to all individual TypeScript configs of the projects that are part of the monorepo workspace.
```

Comment:
```suggestion
This new `tsconfig.json` is the entry point for TypeScript project references, pointing to all individual TypeScript configs of the projects that are part of the monorepo workspace. It includes an explicitly empty `files` array, because it intentionally does not contain any files of its own. This is known as a **solution style config**.
```

We reference solution style config later but never define it. This feels like the logical place to do so.

Reply:
cool. yeah I had it in there but then dropped it again. Didn't want to introduce other terms, but it makes sense I think. thank you